### PR TITLE
Backend/enhancement/forgot-password #265

### PR DESCRIPTION
**Pull request by emresin12**
forgot password functionality implemented.

### CODE REVIEW

File: app/backend/src/main/kotlin/com/gamelounge/backend/service/AccessService.kt [hunk 1]
```diff
@@ -101,4 +106,38 @@ class AccessService (
...
+        //validate username and email
+        val user = userRepository.findByUsername(username)
+            ?: throw UsernameNotFoundException("User not found.")
+        if (user.email != email){
+            throw EmailNotFoundException("Email not found.")
+        }
+        //generate token
+        val token = passwordResetTokenService.createToken(user)
+        //send email
+        emailService.sendSimpleMessage(email, "Password Reset", "Click the link to reset your password: http://localhost:8080/reset-password?token=$token")
```
Comment (ahmetkudu) [inline app/backend/src/main/kotlin/com/gamelounge/backend/service/AccessService.kt:119]: The link to send to the e-mail cannot be a localhost link. This needs to be parameterized. Maybe you can solve it by adding a key value to application.properties.

**Review by ahmetkudu — CHANGES_REQUESTED** [review]
Could you solve the problem I depicted in my review comment?

File: app/backend/src/main/kotlin/com/gamelounge/backend/service/AccessService.kt (hunk 1 above)
Comment (emresin12) [reply app/backend/src/main/kotlin/com/gamelounge/backend/service/AccessService.kt]: ok it is fixed now

**Comment by Legervad** [comment]
It seems fine, works on my local. Merging.

**Pull request merged**